pkg/app/server/grpcapi/web_api.go

```diff
@@ -1008,8 +1008,53 @@ func (a *WebAPI) CancelDeployment(ctx context.Context, req *webservice.CancelDep
 	}, nil
 }
 
-func (a *WebAPI) SkipStage(_ context.Context, _ *webservice.SkipStageRequest) (*webservice.SkipStageResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "not implemented")
+func (a *WebAPI) SkipStage(ctx context.Context, req *webservice.SkipStageRequest) (*webservice.SkipStageResponse, error) {
+	claims, err := rpcauth.ExtractClaims(ctx)
+	if err != nil {
+		a.logger.Error("failed to authenticate the current user", zap.Error(err))
+		return nil, err
+	}
+
+	deployment, err := getDeployment(ctx, a.deploymentStore, req.DeploymentId, a.logger)
+	if err != nil {
+		return nil, err
+	}
+	if claims.Role.ProjectId != deployment.ProjectId {
+		return nil, status.Error(codes.PermissionDenied, "Requested deployment does not belong to your project")
+	}
+	stage, ok := deployment.StageMap()[req.StageId]
+	if !ok {
+		return nil, status.Error(codes.FailedPrecondition, "The stage was not found in the deployment")
+	}
+	if stage.Name != model.StageAnalysis.String() {
```

Comment:
could be `stage.Skippable()` in model

Reply:
Thanks.
I addressed by https://github.com/pipe-cd/pipecd/pull/3529/commits/2810493bebbceef5272c260988a2b9312d516c26